storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -386,13 +406,27 @@ public LogConfig(Map<?, ?> props, Set<String> overriddenConfigs) {
         this.messageFormatVersion = MetadataVersion.fromVersionString(getString(TopicConfig.MESSAGE_FORMAT_VERSION_CONFIG));
         this.messageTimestampType = TimestampType.forName(getString(TopicConfig.MESSAGE_TIMESTAMP_TYPE_CONFIG));
         this.messageTimestampDifferenceMaxMs = getLong(TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG);
+        this.messageTimestampBeforeMaxMs = getMessageTimestampBeforeMaxMs();
+        this.messageTimestampAfterMaxMs = getLong(TopicConfig.MESSAGE_TIMESTAMP_AFTER_MAX_MS_CONFIG);
```

Comment:
Maybe this was called out elsewhere - I will go through the KIP + email discussion - but shouldn't this also default to the specified value of `log.message.timestamp.difference.max.ms` (default or otherwise) if `log.message.timestamp.before.max.ms` is not set?

If not do we not risk the following situation?
* A customer has specified a value of `log.message.timestamp.difference.max.ms` to 100ms.
* The customer upgrades their Kafka version.
* The customer does not set `log.message.timestamp.before.max.ms`.
* There is a misbehaving producer which previously sent records with CreateTimestamp > 10000ms in the future.
* `log.message.timestamp.after.max.ms` defaults to the gigantic number since it is not set and messages from that producer start being accepted while they were previously rejected.

Reply:
Good call out @clolov. To make the transition smooth for customers, we will need to check if customer has changed the value of for both `log.message.timestamp.before.max.ms` and `log.message.timestamp.after.max.ms` and fall back to `log.message.timestamp.difference.max.ms` if it is not set.

I will update with the PR, thank you.